### PR TITLE
Assign variables in .s3cfg file

### PR DESCRIPTION
**Pull request by MiguelSavignano**
To add gpg_passphrase and gpg_command in s3cfg file, assing variables using the tacoma.yml file.

```tacoma.yml
#tacoma.yml
 project:
   aws_identity_file: "/path/to/pem/file/my_project.pem"
   aws_secret_access_key: "YOURSECRETACCESSKEY"
   aws_access_key_id: "YOURACCESSKEYID"
   region: "REGION"
   repo: "$HOME/projects/my_project"
   s3cfg:
     gpg_command: /usr/bin/gpg
     gpg_passphrase: mygpg_passphrase
```

You can override all posible variables in .s3cfg

### CODE REVIEW

**Comment by pantulis** [comment]
what does this do if I have no 's3cfg' section on my tacoma.yml file?   Will it create a new s3cfg file based on the included template with default values?

**Comment by MiguelSavignano** [comment]
yes, if no exists 's3cfg' section in tacoma.yml will create new s3cfg with default values.
```
[default]
access_key = <%= @aws_access_key_id %>
bucket_location = US
cloudfront_host = cloudfront.amazonaws.com
cloudfront_resource = /2010-07-15/distribution
default_mime_type = binary/octet-stream
delete_removed = False
dry_run = False
encoding = UTF-8
encrypt = False
follow_symlinks = False
force = False
get_continue = False
gpg_command = None
gpg_decrypt = %(gpg_command)s -d --verbose --no-use-agent --batch --yes --passphrase-fd %(passphrase_fd)s -o %(output_file)s %(input_file)s
gpg_encrypt = %(gpg_command)s -c --verbose --no-use-agent --batch --yes --passphrase-fd %(passphrase_fd)s -o %(output_file)s %(input_file)s
gpg_passphrase = 
guess_mime_type = True
host_base = s3.amazonaws.com
host_bucket = %(bucket)s.s3.amazonaws.com
human_readable_sizes = False
list_md5 = False
log_target_prefix = 
preserve_attrs = True
progress_meter = True
proxy_host = 
proxy_port = 0
recursive = False
recv_chunk = 4096
reduced_redundancy = False
secret_key = <%= @aws_secret_access_key %>
send_chunk = 4096
simpledb_host = sdb.amazonaws.com
skip_existing = False
socket_timeout = 300
urlencoding_mode = normal
use_https = False
verbosity = WARNING
```

**Comment by pantulis** [comment]
The expected behaviour should be letting the ~/.s3cfg file alone if no s3cfg section is found in the Tacoma file.  Otherwise Tacoma becomes the exclusive tool you would use to manage s3cfg, and that's not its intended use. 

Would you be so kind to adapt your patch for this?

File: lib/template/s3cfg [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
-reduced_redundancy = False
-secret_key = <%= @aws_secret_access_key %>
-send_chunk = 4096
-simpledb_host = sdb.amazonaws.com
-skip_existing = False
-socket_timeout = 300
-urlencoding_mode = normal
-use_https = False
-verbosity = WARNING
+access_key = <%= @s3cfg['access_key'] || @aws_access_key_id %>
```
Comment (pantulis) [inline lib/template/s3cfg]: Change this

**Pull request merged**